demo/src/App.js

```diff
@@ -60,36 +60,16 @@ export default class App extends PureComponent {
       zoom: -5.5,
       target: [30000, 10000, 0]
     };
-    const propSettings = {
-      useTiff: true,
-      imageHeight: source.height * source.tileSize,
-      imageWidth: source.width * source.tileSize,
-      tileSize: source.tileSize,
-      sourceChannels: source.channels,
-      minZoom: Math.floor(
-        -1 *
-          Math.log2(
-            Math.max(
-              source.height * source.tileSize,
-              source.width * source.tileSize
-            )
-          )
-      ),
-      maxZoom: -9
-    };
-    const props = {
-      initialViewState,
-      ...propSettings,
-      ...this.state
-    };
+    const { sliderValues, colorValues, viewHeight, viewWidth } = this.state;
     const sliders = this.sliders.map(sliderObj => {
-      const Slider = Object.values(sliderObj)[0];
+      const ChannelSlider = Object.values(sliderObj)[0];
```

Comment:
Perhaps a dumb question, but is there just one key-value in this object?

Reply:
Yes.